File: src/main/java/net/gtaun/shoebill/data/VehicleState.java
package net.gtaun.shoebill.data;

/**
 * Created by marvin on 02.05.15 in project shoebill-api.
 * Copyright (c) 2015 Marvin Haschker. All rights reserved.
 */
public class VehicleState {

    public int driver, passenger, backLeft, backRight;

    public VehicleState() {
    }

    public VehicleState(int driver, int passenger, int backLeft, int backRight) {
        this.driver = driver;
        this.passenger = passenger;
        this.backLeft = backLeft;
        this.backRight = backRight;
    }

    public int getDriver() {
        return driver;
    }

    public void setDriver(int driver) {
        this.driver = driver;
    }

    public int getPassenger() {
        return passenger;
    }

    public void setPassenger(int passenger) {
        this.passenger = passenger;
    }

    public int getBackLeft() {
        return backLeft;
    }

    public void setBackLeft(int backLeft) {
        this.backLeft = backLeft;
    }

    public int getBackRight() {
        return backRight;
    }

    public void setBackRight(int backRight) {
        this.backRight = backRight;
    }
}

File: src/main/java/net/gtaun/shoebill/data/VehicleState.kt
package net.gtaun.shoebill.data

/**
 * This class should be used with functions that were introduced in SA:MP 0.3.7.
 *
 * @author Marvin Haschker
 */
class VehicleState @JvmOverloads constructor(@JvmField val driver: Int = 0,
                                             @JvmField val passenger: Int = 0,
                                             @JvmField val backLeft: Int = 0,
                                             @JvmField val backRight: Int = 0) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is VehicleState) return false

        if (driver != other.driver) return false
        if (passenger != other.passenger) return false
        if (backLeft != other.backLeft) return false
        if (backRight != other.backRight) return false

        return true
    }

    override fun hashCode(): Int {
        var result = driver
        result = 31 * result + passenger
        result = 31 * result + backLeft
        result = 31 * result + backRight
        return result
    }


}
